### PR TITLE
[ace] tao as feature

### LINKED ISSUE #8447: [New Port Request] ace+tao
Library name: ace+tao

Library description: ACE is a C++ framework for implementing distributed and networked applications, TAO is a C++ implementation of the OMG's CORBA standard.

Source repository URL: https://github.com/DOCGroup/ACE_TAO

Project homepage (if different from the source repository): http://download.dre.vanderbilt.edu/

### PR DESCRIPTION
**Pull request by victorburckel**
Build tao as feature with ace

- What does your PR fix? Fixes #8447. It tries to continue work performed in #8468.

- Which triplets are supported/not supported? Have you updated the CI baseline?
CI baseline was not updated, it should work with all triplets currently supported by ace, which should be all supported triplets once #11369 and #11441 will be merged.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @victorburckel 
Thanks for this PR.

Since this PR is related with #11369 and #11441. 
Personally, it would be better to update the changes after these two PRs are merged.

What do you think about this?

**Comment by victorburckel** [comment]
Hi @NancyLi1013 
Yes sure, there's even #11464 that should be merged first. I will then have to do some small adaptations.

**Comment by NancyLi1013** [comment]
@victorburckel 
Could you please resolve the conflicts?

**Comment by victorburckel** [comment]
Ok I merged the master branch and resolved the conflicts. I updated the CONTROL file to set the version to -4 because it should be merged after #11441 (which updates the version to -2) and #11464 (which updates the version to -3). I guess there will still be conflicts when the both will merged, especially the last one which uses arrays to enumerate includes to install for windows build.

File: ports/ace/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: ace
-Version: 6.5.9-2
+Version: 6.5.9-4
```
Comment (NancyLi1013) [inline ports/ace/CONTROL:2]:
```suggestion
Version: 6.5.9-3
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/ace/CONTROL (hunk 1 above)
Comment (victorburckel) [reply ports/ace/CONTROL:2]: There's still the 3rd and last PR #11464 waiting to merged which will update version to 6.5.9-3 that's why I set it to 6.5.9-4. Do your still want me to update the version (and maybe update it later when resolving the conflicts) ?

Comment (victorburckel) [reply ports/ace/CONTROL]: I did the adaptations required by #11464 and updated the version. If #11464 is merged first,the CONTROL file will conflict and I'll update it back to -4. If this PR is merged first I guess the other PR can be closed as its changes are now included in this PR

**Comment by JackBoosY** [comment]
Now, I think you can continue this PR.

**Comment by victorburckel** [comment]
Thanks, conflicts resolved.

**Comment by victorburckel** [comment]
Hi, are you waiting for any feedback, rework from me ?

**Comment by NancyLi1013** [comment]
@victorburckel 

Could you please resolve the conflicts?

All features have passed with the following triplets:
- x86-windows
- x64-windows
- x64-windows-static

**Review by NancyLi1013 — APPROVED** [review]

**Comment by victorburckel** [comment]
Ok thanks for the test, I merged master and adapt to reflect the changes made to the portfile, and updated the version.
Seems like multiple packages fail to build currently.

**Comment by JackBoosY** [comment]
Please wait for us until the baseline error is resolved.

**Comment by victorburckel** [comment]
Hi, I can't see if build failures are related to this PR or not. Can I do something to help making progress on that PR ?

**Comment by NancyLi1013** [comment]
@victorburckel 
The failures are not related with PR.  We will rerun it once resolved in #11839.

File: ports/ace/portfile.cmake [hunk 2]
```diff
@@ -104,137 +112,224 @@ endif()
...
-          DESTINATION ${CURRENT_PACKAGES_DIR}/lib
-      )
-    endforeach()
-  endfunction()
+    set(ACE_PROTOCOLS_INCLUDE_FOLDERS "ace/HTBP" "ace/INet" "ace/RMCast"
+                                      "ace/TMCast")
+    install_includes(${ACE_ROOT}/protocols "${ACE_PROTOCOLS_INCLUDE_FOLDERS}")
 
-  set(ACE_LIBRARIES "ACE" "ACE_Compression" "ACE_ETCL" "ACE_ETCL_Parser" "ACE_Monitor_Control" "ACE_QoS" "ACE_RLECompression")
-  install_libraries(${ACE_ROOT} "${ACE_LIBRARIES}")
```
Comment (dan-shaw) [inline ports/ace/portfile.cmake:167]: Why are we removing install_libraries? Also pinging the library author @jwillemsen for any feedback on this PR

Comment (victorburckel) [reply ports/ace/portfile.cmake:167]: Actually these libraries do not contain any export, so keeping them makes the build fail in post checks.

Comment (jwillemsen) [reply ports/ace/portfile.cmake:167]: I lack time to test this, I assume @victorburckel has tested this and builds his application based on the vcpkg packaged ace/tao

**Comment by dan-shaw** [comment]
Thanks for the PR!

**Pull request merged**